File: C++ solutions/2447. Number of Subarrays With GCD Equal to K.cpp
class Solution {
public:
    int GCD(int A, int B) {
        if(B==0) return A;
        return GCD(B, A%B);
    }
    int subarrayGCD(vector<int>& nums, int k) {
        int ans=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==k) ans++;
            int gcd=nums[i];
            for(int j=i+1;j<nums.size();j++){
                int res=GCD(gcd,nums[j]);
                if(res==k) ans++;
                gcd=res;
            }
        }
        return ans;
    }
};

File: JavaScript solutions/2447. Number of Subarrays With GCD Equal to K.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarrayGCD = function (nums, k) {
  const GCD = (A, B) => {
    if (B == 0) return A;
    return GCD(B, A % B);
  };

  let ans = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == k) ans++;
    let gcd = nums[i];
    for (let j = i + 1; j < nums.length; j++) {
      let res = GCD(gcd, nums[j]);
      if (res == k) ans++;
      gcd = res;
    }
  }
  return ans;
};
